src/app/webui/components/Playground/PlaygroundView.tsx

```
@@ -0,0 +1,571 @@
+                                    <pre className="whitespace-pre-wrap text-xs bg-muted/50 p-2 rounded-sm font-mono">{typeof otherPart === 'object' ? JSON.stringify(otherPart, null, 2) : String(otherPart)}</pre>
+                                  </div>
+                                ))
+                              )}
+                            </div>
+                          );
+                        } else {
+                          return <p className="text-xs text-destructive">Image data is not in a recognized format. Data: {JSON.stringify(toolResult.data)}</p>;
+                        }
+                      })()
+                    ) : toolResult.metadata?.type === 'markdown' && toolResult.data ? (
+                        <div className="prose prose-sm dark:prose-invert max-w-none p-3 bg-background rounded border border-input leading-relaxed" dangerouslySetInnerHTML={{ __html: typeof toolResult.data === 'string' ? toolResult.data.replace(/\n/g, '<br/>') : JSON.stringify(toolResult.data, null, 2) }} />
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)